Added CITs for Circle processor.

cuff-links opened the pull request:
Closes #100106

## Summary

This PR is to add test coverage for the Ingest Pipelines Circle Processor. 

**Test cases being covered**

1. Prevents submission 0/2 required fields given.
2. Saves processor with default options
3. Saves processor with setting optional options.

Commit:
Added CITs for Circle processor.

elasticmachine commented:
Pinging @elastic/kibana-stack-management (Team:Stack Management)

cuff-links commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'master' into CIT_For_Circle_Processor

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130421/)
* Commit: bfe2dd3659a5ac39e7be55778235611374823fda
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types and Public API Docs](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130421/execution/node/428/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130421/","number":"130421","commit":"bfe2dd3659a5ac39e7be55778235611374823fda"}]}
PIPELINE-->

Commit:
Fixed issue with form function using int instead of string.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130470/)
* Commit: a25084fb95c1f49e00f69fc2e55be2631269ca85
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-101871/a25084fb95c1f49e00f69fc2e55be2631269ca85)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/45ed0035b6272f1873cbf19472ec871a00c5f65c) | [after](https://github.com/elastic/kibana/commit/a25084fb95c1f49e00f69fc2e55be2631269ca85) | diff |
| --- | --- | --- | --- |
| `ingestPipelines` | 687.5KB | 687.6KB | +76.0B |

### History
* :broken_heart: [Build #130421](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130421/) failed bfe2dd3659a5ac39e7be55778235611374823fda

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130421/","number":"130421","commit":"bfe2dd3659a5ac39e7be55778235611374823fda"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130470/","number":"130470","commit":"a25084fb95c1f49e00f69fc2e55be2631269ca85"}]}
PIPELINE-->

cuff-links commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'master' into CIT_For_Circle_Processor

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130774/)
* Commit: f1a1a3e42a7bf52cf25018420a49a4fcf5934df6
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-101871/f1a1a3e42a7bf52cf25018420a49a4fcf5934df6)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/012bb4490b5598790bc88fd112ab1c3d006a9a61) | [after](https://github.com/elastic/kibana/commit/f1a1a3e42a7bf52cf25018420a49a4fcf5934df6) | diff |
| --- | --- | --- | --- |
| `ingestPipelines` | 687.5KB | 687.6KB | +76.0B |

### History
* :green_heart: [Build #130470](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130470/) succeeded a25084fb95c1f49e00f69fc2e55be2631269ca85
* :broken_heart: [Build #130421](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130421/) failed bfe2dd3659a5ac39e7be55778235611374823fda

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130421/","number":"130421","commit":"bfe2dd3659a5ac39e7be55778235611374823fda"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130470/","number":"130470","commit":"a25084fb95c1f49e00f69fc2e55be2631269ca85"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130774/","number":"130774","commit":"f1a1a3e42a7bf52cf25018420a49a4fcf5934df6"}]}
PIPELINE-->

Commit:
[Discover] Deangularization of search embeddable (#100552)

* [Discover] Render empty embeddable

* First version of grid embeddable

* More search embeddable

* Almost stable version

* Fixing typescript errors

* Fixing filtering and sorting

* Add data-shared-item to DiscoverGridEmbeddable

* Trigger rerender when title changes

* Fixing incorrectly touched files

* Remove search_embeddable

* Remove lodash

* Fixing imports

* Minor fixes

* Removing unnecessary files

* Minor fix

* Remove unused import

* Applying PR comments

* Applying PR comments

* Removing search embeddable

* Fix missing import

* Addressing PR comments

* Do not memoize saved search component

* Applying Matthias's suggestion

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

stratoula committed:
[Dashboard]: Fixes disabled viz filter is applied (#101859)

* Fixes filter is applied even if is disabled on a dashboard

* Fix 18n problem

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

dasansol92 committed:
[Security solution][Endpoint] Removes zip compression when creating artifacts (#101379)

* Removes zlib compression when creating artifacts. Also fixes related unit tests and removes old code

* Replaces artifact in new manifest using the ones from fleet client with zlip compression

* Fixes create_policy_artifact_manifest pushArtifacts missing new manifest. Also fixes unit tests

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

justinkambic committed:
[Uptime] Add owner and description properties to Uptime's kibana.json (#101963)

Co-authored-by: Shahzad <shahzad31comp@gmail.com>
Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

jgowdyelastic committed:
[ML] Switching to new datafeed preview (#101780)

* [ML] Switching to new datafeed preview

* fixing wizard test button

* adding schema validator

* fixing tests

* adding check for empty detectors list

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

mistic committed:
skip flaky suite (#100438)

streamich committed:
Management app locator (#101795)

* feat: 🎸 create management app locator

* refactor: 💡 simplify management locator

* feat: 🎸 export management app locator from plugin contract

* feat: 🎸 improve share plugin exports

* test: 💍 fix test mock

* test: 💍 adjust test mocks

* Update src/plugins/management/public/plugin.ts

Co-authored-by: Tim Roes <mail@timroes.de>

* Update src/plugins/management/public/types.ts

Co-authored-by: Tim Roes <mail@timroes.de>

* Update src/plugins/management/public/types.ts

Co-authored-by: Tim Roes <mail@timroes.de>

* Update src/plugins/management/server/plugin.ts

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Tim Roes <mail@timroes.de>

azasypkin committed:
Upgrade `polished` dependency (`1.9.2` → `3.7.2`, `4.0.5` → `4.1.3`). (#101719)

jloleysens committed:
move the example app to be mounted in the developerExamples plugin instead of visible in navbar (#101464)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

gmmorris committed:
[Task Manager] adds capacity estimation to the TM health endpoint (#100475)

Adds Capacity Estimation to the Task Manager Health Endpoint.
Below is a diagram depicting what information we use to estimate the varying capacity variables.

Please use the user facing docs to understand how it fits together. If the docs aren't clear enough - make a review comment and I'll clarify in the docs.

pheyos committed:
[ML] Functional tests - stabilize alerting flyout test (#102030)

This PR stabilizes and re-activates the ML alerting flyout test suite.

dgieselaar committed:
[APM] Display automatic deployment annotations correctly (#102020)

ashokaditya committed:
[Security Solution][Endpoint] Actions Log API (#101032)

* WIP

add tabs for endpoint details

* fetch activity log for endpoint

this is work in progress with dummy data

* refactor to hold host details and activity log within endpointDetails

* api for fetching actions log

* add a selector for getting selected agent id

* use the new api to show actions log

* review changes

* move util function to common/utils

in order to use it in endpoint_hosts as well as in trusted _apps

review suggestion

* use util function to get API path

review suggestion

* sync url params with details active tab

review suggestion

* fix types due to merge commit

refs 3722552f739f74d3c457e8ed6cf80444aa6dfd06

* use AsyncResourseState type

review suggestions

* sort entries chronologically with recent at the top

* adjust icon sizes within entries to match mocks

* remove endpoint list paging stuff (not for now)

* fix import after sync with master

* make the search bar work (sort of)

this needs to be fleshed out in a later PR

* add tests to middleware for now

* use snake case for naming routes

review changes

* rename and use own relative time function

review change

* use euiTheme tokens

review change

* add a comment

review changes

* log errors to kibana log and unwind stack

review changes

* search on two indices

* fix types

* use modified data

* distinguish between responses and actions and respective states in UI

* use indices explicitly and tune the query

* fix types after sync with master

* fix lint

* do better types

review suggestion

* add paging to API call

* add paging info to redux store for activityLog

* decouple paging action from other API requests

* use a button for now to fetch more data

* add index to fleet indices

else we get a type check error about the constant not being exported correctly
from `x-pack/plugins/fleet/common/constants/agent`

* add tests for audit log API

* do semantic paging from first request

* fix ts error

review changes

* add document id and total to API

review suggestions

* update test

* update frontend to consume the modified api correctly

* update mock

* rename action

review changes

* wrap mock into function to create anew on each test

review changes

* wrap with schema.maybe and increase page size

review changes

* ignore 404

review changes

* use i18n

review changes

* abstract logEntry component logic

review changes

* move handler logic to a service

review changes

* update response object

review changes

* fix paging to use 50 as initial fetch size

* fix translations and move custom hook to component file

review changes

* add return type

review changes

* update default value for page_size

review changes

* remove default values

review changes

https://github.com/elastic/kibana/tree/master/packages/kbn-config-schema#schemamaybe

https://github.com/elastic/kibana/tree/master/packages/kbn-config-schema#default-values

* fix mock data

refs 1f9ae7019498c616455f1109a524741e212106ca

* add selectors for data

review changes

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

shahzad31 committed:
[Uptime] Align synthetics report types (#101855)

alisonelizabeth commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_editor/__jest__/processors/circle.test.tsx:
```diff
@@ -0,0 +1,112 @@
+    // Click submit button with only the type defined
+    await saveNewProcessor();
+
+    // Expect form error as "field" is required parameter
```
```suggestion
    // Expect form error as "field" and "shape_type" are required parameters
```

cuff-links replied:
Fixed.

yuliacech commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_editor/__jest__/processors/circle.test.tsx:
```diff
@@ -0,0 +1,112 @@
+
+    expect(processors[0].circle).toEqual({
+      field: 'field_1',
+      error_distance: 1,
```
I noticed that `error_distance` has a default value of 1 in the UI, but it's not mentioned in the docs. I think if `error_distance` doesn't have a default, then its input field should rather stay empty and force the user to set a value. We might need a bug issue for that, but I'd like to confirm with @alisonelizabeth first.

alisonelizabeth replied:
Great catch @yuliacech! I didn't notice your review until after mine (I left a similar comment) 😄. 

@danhermann can you help us understand how the `error_distance` option should behave for the `circle` processor? The docs don't mention any default value, but we provide a default value of `1` in the UI. Does this make sense, or should we always force the user to provide their own value?

danhermann replied:
I don't know enough about how the underlying circle shape works to know whether a default value (perhaps scaled by the radius of the circle) for `error_distance` would make sense there or not. Perhaps someone from @elastic/es-analytics-geo make a suggestion?

imotov replied:
A default value would only make sense if it is scaled to the radius of a circle. Unfortunately, we are setting up processor at this point and have no information about what type of circles the user is going to ingest. So, I think it would be the best to leave it blank.

alisonelizabeth replied:
Thanks @imotov for clarifying! I opened https://github.com/elastic/kibana/issues/102147 to address on our side.

yuliacech commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_editor/__jest__/processors/circle.test.tsx:
```diff
@@ -0,0 +1,112 @@
+    // Add "target_field" value
+    form.setInputValue('targetField.input', 'target_field');
+
+    form.setInputValue('errorDistanceField.input', '10');
```
As I mentioned in the previous comment, `error_distance` has a default value set in the UI, but the parameter itself is required. So it should actually be tested in the first 2 tests.

cuff-links replied:
Sent the parameter. #102147 will add test to check for the error once that's implemented.

Dosant committed:
[Index Patterns] Cover field editor with a11y tests (#101888)

renovate[bot] committed:
Update dependency @elastic/charts to v30.1.0 (master) (#101683)

alisonelizabeth commented on x-pack/plugins/ingest_pipelines/public/application/components/pipeline_editor/__jest__/processors/circle.test.tsx:
```diff
@@ -0,0 +1,112 @@
+    await saveNewProcessor();
+
+    // Expect form error as "field" is required parameter
+    expect(form.getErrorsMessages()).toEqual([
```
It looks like [`error_distance` should also be a required option](https://www.elastic.co/guide/en/elasticsearch/reference/master/ingest-circle-processor.html). I see the default value is set to `1` in the UI, which is why there is no error message here. But if you remove the value it sets it to `0` on save, rather than showing a field error. I don't know enough about the `error_distance` option to say whether this would be expected behavior or not. It may be worth reaching out to an ES engineer to ask though. It might be safer to show a field error, in which case we should open up a separate issue to address.

alisonelizabeth replied:
See comment thread above. I opened https://github.com/elastic/kibana/issues/102147 to address.

cuff-links replied:
OK. So I went ahead and set the error_distance in the two tests where we are setting parameters. I see the comment in #102147 where you said you will update the behavior to validate when the parameter is not sent in the first test. So I will leave that to you and get this merged. @alisonelizabeth

alisonelizabeth reviewed (approved):
Nice work @cuff-links! LGTM. I left one nit as well as a comment on how the `error_distance` option should behave.

alisonelizabeth committed:
[ILM] Migrate to new page layout (#101927)

Commit:
Update building_a_plugin.mdx (#101921)

* Update building_a_plugin.mdx

* put back the numbers on the same line

* Add info about requiredBundles

* Fix numbers again

* Update dev_docs/tutorials/building_a_plugin.mdx

Co-authored-by: Mikhail Shustov <restrry@gmail.com>

* Update dev_docs/tutorials/building_a_plugin.mdx

Co-authored-by: Mikhail Shustov <restrry@gmail.com>

Co-authored-by: Mikhail Shustov <restrry@gmail.com>
Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

joshdover committed:
Fix delayed status API updates in alerting and task_manager (#101778)

sorenlouv committed:
[APM] Add AWS and Azure icons for additional services (#101901)

legrego committed:
Document platform security plugins in kibana.json (#101965)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

nchaulet committed:
[Fleet] Display NOTICE.txt from package if it exists (#101663)

formgeist committed:
[APM] Change impact indicator size (#102060)

dgieselaar committed:
[APM] Enforce span creation/naming for ES searches (#101856)

formgeist committed:
[APM] Change View full trace button fill (#102066)

ecezalp committed:
[Security Solution] Refactor rules for modularization by updating bulkCreate and wrapHits methods (#101544)

creates bulkCreate and wrapHits factories for the modularization of the detection engine

Co-authored-by: Marshall Main <marshall.main@elastic.co>

kertal committed:
[Discover] Unskip runtime field editor test (#101059)

Kerry350 committed:
[Logs / Metrics UI] Convert logs and metrics pages to the new Observability page template (#101239)

* Convert Logs and Metrics pages to use the Observability page template

brianseeders committed:
[FTSR] Add pluginFunctional to test suites (#102113)

cavokz committed:
Fix esArchiver path in the Jenkins context (#102095)

ajosh0504 committed:
[ML] Adds Authentication module with six ML jobs for ECS data (Auditbeat, Winlogbeat, Filebeat and Logs) (#101840)

* Adding Security Authentication jobs in 7.14

* Renamed some jobs

* Changing memory limits and linting change

* Linting fix

* Changed the order

* Adding module to ml_modules.tsx

* Update recognize_module.ts

this test modules uses older Auditbeat data which predates the event.category field so the test has to be skipped per https://elastic.zoom.us/j/93000943632?pwd=TmpvNWhtYUNzMUc0c0N6Tlc2QlVPZz09

* Update recognize_module.ts

needs to be a single line

* Update recognize_module.ts

Some linters want spaces and some linters want no spaces. This linter wants spaces.

* descriptions

added description text

* Update auth_rare_hour_for_a_user.json

removed a wayward newline char

* Minor nitpicking

* memory limits

raised memory limits to 128mb which is larger than the highest observed peak model bytes for the most memory hungry jobs in this event class.

Co-authored-by: Craig <mailredirector36@gmail.com>
Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

ppisljar committed:
adds support for setting multiple variables to varSet (#100458)

Commit:
[kbnArchiver] handle archives which have \r\n (#102118)

Co-authored-by: spalger <spalger@users.noreply.github.com>

Commit:
[Maps] Fix geo alerts handling of multi-fields (#100348)

cjcenizal committed:
Discourage use of legacy index templates (#101533)

* Hide legacy index templates table if the user doesn't have any.
* Render deprecation warning above legacy index templates table and in legacy index template wizard.
* Update index template doc link to point to the new docs.

ymao1 committed:
[Alerting][Docs] Reformatting rule types docs (#101420)

* Reformatting rule types docs

* Apply suggestions from code review

Co-authored-by: gchaps <33642766+gchaps@users.noreply.github.com>

* Restructure geo rule page

* Apply suggestions from code review

Co-authored-by: gchaps <33642766+gchaps@users.noreply.github.com>

* PR fixes

Co-authored-by: gchaps <33642766+gchaps@users.noreply.github.com>

stratoula committed:
[Tagcloud] Replaces current implementation with elastic-charts (#100017)

* WIP - Replace tagcloud with es-charts wordcloud

* Cleanup and add unit tests

* Fix interpreter test

* Update all tagcloud snapshots

* Partial fix tagcloud test

* Fix some other functional tests, add migration script, update sample data

* Replace getColor with getCategorixalColor

* Fix functional test

* Apply clickhandler event for filtering by clicking the word

* Fix weight calculation

* Add a unit test and fix functional

* Change the cursor to pointer

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

kertal committed:
[Discover] Improve functional test wording for responsive tests(#101744)

jloleysens committed:
[Fleet] Adjust add integration flow (#101714)

* Initial commit, very WIP

- added link to the integrations UI instead of add integration
  in fleet
- added a new piece of route state that gets passed to integration
  from fleet when browsing integrations for a policy
- when the integration is being added the page will send the user
  back to the policy overview instead of back to integrations UI

* remove unnecessary agent policy clear function

* added # to path so that navigation is correctly handled

* added logic to read the forward agent policy id

* remove inline select integration package from fleet add integration

* updated toast notification

* using query parameter to pass policy id back to create policy package page

* removed policyId from route path

* fix type issue

* updated the select agent field layout per the designs

* simpified item rendering in combobox and fixed combobox z-index issue

* added comment

* fix types and i18n

* updated icon and removed unused i18n

* refactor to using styled components for cusomt z-index styling

* attempt to fix integration test

* added scroll functionality for dealing with fixed footers that might be obstructing content

* fix scroll direction!

* attempting another scroll algorithm

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

flash1293 committed:
rename beta badge (#101884)

flash1293 committed:
[Lens] do not allow custom bounds on percentage chart (#101743)

flash1293 committed:
reduce inner timeout (#102081)

formgeist committed:
[Observability] Fixes border shadow and other enhancements (#102072)

* [Observability] Fixes border shadow and other enhancements

* Update x-pack/plugins/observability/public/components/app/section/index.tsx

Co-authored-by: Cauê Marcondes <55978943+cauemarcondes@users.noreply.github.com>

* [Observability] Remove unused import and formatting

Co-authored-by: Cauê Marcondes <55978943+cauemarcondes@users.noreply.github.com>
Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

dej611 committed:
[Lens]Show dynamic coloring only for metrics (#101777)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

dej611 committed:
[Lens] Table headers are aligned the same as cells (#101875)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

formgeist committed:
[Observability] Change sortkey navigation in APM and Uptime (#102134)

* [APM] Change navigation registry sortkey

* [Uptime] Change navigation registry sortkey

thomasneirynck committed:
[Maps] Remove jitter from bounds-coordinates (#100717)

flash1293 committed:
[Lens] Add reset and hide actions to columns on dashboard (#102025)

mattkime committed:
[Index Patterns] Move rollup config to index pattern management (#102145)

* move rollup config to index pattern management

Commit:
[Uptime] Skip flaky test (#102163)

shahzad31 committed:
[User Experience App] Update to make sure env filter is being used (#102063)

ashokaditya committed:
[Security Solution][Endpoint] Actions log entries display relative time only up to 1 hr from now (#102162)

* show relative time only upto an hr from now

fixes elastic/security-team/issues/1313

* Update formatted_date_time.tsx

undo changes done in e38e202e81f8dd2796bebe9069175d31052c52f2

jloleysens committed:
[Screenshot Mode] Prevent newsfeed plugin from fetching items (#101256)

* - updated newsfeed to depend on low-level screenshotmode plugin
- added new NeverFetch driver that will never resolve to true
  for new fetch requests
- added plugin-level test for expected behaviour

* added screenshotmode contract to start!

* address pr feedback

* removed old test and updated test description

* update use of jest.fn to tidy up code

* simplify jest tests and remove unnecessary async

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

nreese committed:
[Maps] Handle indices without geoip schema defined in mapping (#100487)

* ensure sourceGeoField and destGeoField for docs included in pew pew source

* tslint

flash1293 committed:
background session limitation docs (#102050)

dej611 committed:
[Lens] Keep filters when transitioning from calculation to regular operation (#101872)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

d1v1b committed:
[Fleet] Remove release messaging (#101450)

Co-authored-by: kobarei <koba.reiji@gmail.com>
Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

jgowdyelastic committed:
[ML] Improvements to job saved object sync (#101899)

* [ML] Improvements to job saved object sync

* refactor

* adding testsadding testsadding testsadding testsadding testsadding testsadding testsadding tests

* updating test label

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

dasansol92 committed:
Fixes empty page when removing last element. Fixes wrong event filters count number. Also adds unit tests (#102173)

legrego committed:
Optimize privilege response validation (#102135)

sabarasaba committed:
[Ingest Node Pipelines] Migrate to new page layout (#101894)

* migrate pages to new layout

* fix linter errors

* update translation files

* Nicer try-again cta

* Fix lang and prettier

* small CR changes

* small linter fixes

* fix test copy

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

sabarasaba committed:
[Remote Clusters] Migrate to new page layout (#102042)

* change page templates

* Small copy tweaks

* Update docs

* small CR changes

* exit function call if fails

* update i18n strings

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

joeypoon committed:
[Security Solution] restyle endpoint details flyout (#102092)

byronhulcher committed:
App Search: Domains Table for Crawler Overview (#101515)

* Add delete route for crawler domains

* Added deleteDomain action to CrawlerOverviewLogic

* New DomainsTable component

* Replace EuiCode with DomainsTable in CrawlerOverview

* Fixing mock imports

* Fix spleling

* Fix i18n tag

* Clean up DomainsTable

* Clean up DomainsTable tests

* Clean up CrawlerOverview tests

* Clean up i18n tag in CrawlerOverviewLogic

* Utilize Kibana i18n and React for timestamp display

* Use elastic.co domain in tests

* Remove unneeded type casting

* Adjusting isDateToday check

* Apply suggestions from code review

* Update x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/crawler_overview_logic.ts

Co-authored-by: Constance <constancecchen@users.noreply.github.com>

Dosant committed:
Convert AppServices management pages to KibanaPageTemplate (#101779)

machadoum committed:
[Core] Update getUrlForApp to support deepLinks (#102172)

* [Core] Update getUrlForApp to support deepLinks

* Add unit tests to getUrlForApp with deepLinkId parameter

scottybollinger committed:
[Enterprise Search] Refactor Role mappings to use single endpoint  (#102096)

* Add tooltip back to table row

This was missed when refactoring the table to an EUI component. Built-in mappings have tooltips and don’t have IDs and need to show tooltips instead of actions.

* Fix roleType display

Also missed in the refactor. Made a mistake when copying/pasting

* Refactor logic files to use single endpoint for UI props

As a part of this feature, we are now passing all props needed for the UI in the list endpoint. Previously, whether creating a new mapping, or updating an existing mapping, an endpoint had to be called to fetch the data needed for display. Now all this data comes from the initial fetching of mappings and the other endpoints are no longer needed.

* Refactor WS test to match AS

There was an issue where 100% test coverage was not achieved in Workplace Search. This had already been fixed in App Search by refactoring. This changes the code and test in Workplace Search to match

* Remove server routes

alisonelizabeth committed:
[Snapshot + Restore] Migrate to new page layout structure (#101811)

dmlemeshko committed:
[deps] update chromedriver to 91 (#102199)

tylersmalley committed:
skip flaky suite (#102012)

marius-dr committed:
[QA]Skip of metricbeat dashboard test (#97174)

* fix flakyness of metricbeat dashboard test

* skip test for now

* Update _metricbeat_dashboard.js

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

Commit:
Update aggregation reference docs for 7.14 (#101931)

* Update aggregation reference docs for 7.13

* Add more reference about filtered metrics

* [Lens] Document updates for 7.14

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

yakhinvadim committed:
[Workplace Search] Fix sidebar copy for Account Settings page (#102110)

* Rename PrivateSourcesLayout to PersonalDashboardLayout

The renaming was required because this layout is also used for Account Settings.

* Rename imports

* Rename files

* More personal_dashboard_layout from sources folder to layout folder

* Extract PrivateSourcesSidebar from PersonalDashboardLayout

This is needed to add the ability to pass AccountSettingsSidebar in the
next commit.

* Add and use AccountSettingsSidebar

* Move related styles to personal_dashboard_layout folder

* Move tests from personal_dashboard_layout to private_sources_sidebar

* Add tests for account_settings_sidebar

* Remove redundant import

* Move source-related sidebar text constants to top-level constants file

Now all personal dashboard sidebar copy is in one place

patrykkopycinski committed:
[Osquery] Add support for platform and version fields (#101835)

shahzad31 committed:
[Lens] Export lens save modal (#100381)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

TinaHeiligers committed:
[SO-migrations] fix flaky integration test: Uses new fixture with default distribution (#102139)

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

smith committed:
Flyouts for APM alert annotations (#101632)

bhavyarm committed:
updating the saved objects import file to include by value panels and drilldowns (#102114)

Commit:
Added changed per nits.

rylnd committed:
[Security Solution][CTI] Event enrichment search strategy (#101553)

* Adding boilerplate for new CTI search strategy type

This is going to be a subtype of the general SecSol search strategy;
the main functionality is going to be:

* transformation of the incoming parameters into named equivalents
* transformation of responses to include enrichment context fields
  (matched.*)

* More boilerplate, including tests

A few type errors because our functions don't actually do anything yet,
nor are our request/response types fleshed out.

* Starting to flesh out the request parsing

* Defines a basic request, along with a mock
* Defines helper function to generate should clauses from field values
* Adds placeholder tests throughout

* Fleshing out unit tests around our enrichment query

* Fleshing out response parsing of eventEnrichment strategy

* Fix types from elasticsearch

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

elasticmachine commented:
Pinging @elastic/apm-ui (Team:apm)

elasticmachine commented:
Pinging @elastic/uptime (Team:uptime)

cuff-links commented:
Something went wrong here and will be reissuing this PR.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131537/)
* Commit: 38e0c5003901335ed24c459c0fea99e709a5e458
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-101871/38e0c5003901335ed24c459c0fea99e709a5e458)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/012bb4490b5598790bc88fd112ab1c3d006a9a61) | [after](https://github.com/elastic/kibana/commit/38e0c5003901335ed24c459c0fea99e709a5e458) | diff |
| --- | --- | --- | --- |
| `apm` | 1600 | 1534 | -66 |
| `canvas` | 1089 | 1090 | +1 |
| `discover` | 377 | 368 | -9 |
| `enterpriseSearch` | 1423 | 1438 | +15 |
| `esUiShared` | 152 | 153 | +1 |
| `fleet` | 475 | 476 | +1 |
| `indexManagement` | 517 | 518 | +1 |
| `indexPatternManagement` | 162 | 167 | +5 |
| `infra` | 967 | 902 | -65 |
| `lens` | 687 | 689 | +2 |
| `management` | 34 | 35 | +1 |
| `ml` | 1698 | 1699 | +1 |
| `newsfeed` | 14 | 15 | +1 |
| `observability` | 313 | 241 | -72 |
| `osquery` | 151 | 212 | +61 |
| `rollup` | 169 | 165 | -4 |
| `securitySolution` | 2242 | 2181 | -61 |
| `uptime` | 614 | 553 | -61 |
| `visTypeTagcloud` | 30 | 24 | -6 |
| total |  |  | -254 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/012bb4490b5598790bc88fd112ab1c3d006a9a61) | [after](https://github.com/elastic/kibana/commit/38e0c5003901335ed24c459c0fea99e709a5e458) | diff |
| --- | --- | --- | --- |
| `core` | 1052 | 1053 | +1 |
| `data` | 3266 | 3270 | +4 |
| `esUiShared` | 86 | 88 | +2 |
| `expressions` | 1467 | 1469 | +2 |
| `fleet` | 1006 | 1010 | +4 |
| `indexPatternManagement` | 45 | 0 | -45 |
| `lens` | 161 | 162 | +1 |
| `management` | 38 | 39 | +1 |
| `observability` | 195 | 198 | +3 |
| `screenshotMode` | 16 | 17 | +1 |
| `share` | 61 | 68 | +7 |
| total |  |  | -19 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/012bb4490b5598790bc88fd112ab1c3d006a9a61) | [after](https://github.com/elastic/kibana/commit/38e0c5003901335ed24c459c0fea99e709a5e458) | diff |
| --- | --- | --- | --- |
| `apm` | 4.3MB | 4.2MB | -23.2KB |
| `canvas` | 1.3MB | 1.3MB | +1.2KB |
| `crossClusterReplication` | 293.5KB | 294.2KB | +714.0B |
| `dashboard` | 221.4KB | 222.1KB | +714.0B |
| `dataEnhanced` | 159.6KB | 158.9KB | -707.0B |
| `discover` | 426.8KB | 416.9KB | -9.8KB |
| `enterpriseSearch` | 2.1MB | 2.1MB | +5.4KB |
| `fileUpload` | 784.8KB | 787.6KB | +2.8KB |
| `fleet` | 710.7KB | 704.9KB | -5.8KB |
| `graph` | 1.1MB | 1.1MB | +714.0B |
| `indexLifecycleManagement` | 249.2KB | 248.6KB | -560.0B |
| `indexManagement` | 1.3MB | 1.3MB | +3.6KB |
| `indexPatternManagement` | 552.8KB | 550.9KB | -1.9KB |
| `infra` | 1.8MB | 1.7MB | -25.7KB |
| `ingestPipelines` | 687.5KB | 687.2KB | -330.0B |
| `lens` | 1.4MB | 1.5MB | +26.2KB |
| `licenseManagement` | 123.3KB | 124.0KB | +714.0B |
| `maps` | 2.6MB | 2.6MB | +1.5KB |
| `ml` | 5.9MB | 5.9MB | +1.3KB |
| `observability` | 483.6KB | 451.3KB | -32.3KB |
| `osquery` | 810.5KB | 840.7KB | +30.2KB |
| `remoteClusters` | 163.8KB | 162.9KB | -841.0B |
| `reporting` | 33.7KB | 33.1KB | -605.0B |
| `rollup` | 229.5KB | 230.2KB | +723.0B |
| `security` | 782.3KB | 782.3KB | +67.0B |
| `securitySolution` | 6.9MB | 6.9MB | -10.3KB |
| `snapshotRestore` | 459.5KB | 458.9KB | -595.0B |
| `stackAlerts` | 662.9KB | 663.1KB | +187.0B |
| `uptime` | 1.1MB | 1.1MB | -15.2KB |
| `visTypeTagcloud` | 268.6KB | 17.8KB | -250.8KB |
| total |  |  | -302.8KB |

#### Public APIs missing exports
> Total count of every type that is part of your API that should be exported but is not. This will cause broken links in the API documentation system. Target amount is 0. `Run node scripts/build_api_docs --plugin [yourplugin] --stats exports` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/012bb4490b5598790bc88fd112ab1c3d006a9a61) | [after](https://github.com/elastic/kibana/commit/38e0c5003901335ed24c459c0fea99e709a5e458) | diff |
| --- | --- | --- | --- |
| `indexPatternManagement` | 3 | 0 | -3 |
| `management` | 4 | 5 | +1 |
| `share` | 4 | 6 | +2 |
| total |  |  | -0 |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/012bb4490b5598790bc88fd112ab1c3d006a9a61) | [after](https://github.com/elastic/kibana/commit/38e0c5003901335ed24c459c0fea99e709a5e458) | diff |
| --- | --- | --- | --- |
| `core` | 410.5KB | 410.7KB | +199.0B |
| `data` | 840.1KB | 840.3KB | +238.0B |
| `discover` | 74.5KB | 74.7KB | +227.0B |
| `esUiShared` | 189.8KB | 191.8KB | +2.0KB |
| `expressions` | 202.7KB | 202.8KB | +80.0B |
| `fleet` | 434.8KB | 435.8KB | +1.0KB |
| `indexPatternManagement` | 12.9KB | 19.5KB | +6.6KB |
| `infra` | 140.6KB | 141.3KB | +720.0B |
| `ingestPipelines` | 23.7KB | 23.5KB | -152.0B |
| `kbnUiSharedDeps-elastic` | 2.6MB | 2.6MB | +3.3KB |
| `kbnUiSharedDeps-js` | 6.4MB | 6.4MB | +87.0B |
| `lens` | 38.1KB | 40.0KB | +1.9KB |
| `management` | 15.6KB | 16.2KB | +678.0B |
| `ml` | 66.1KB | 65.8KB | -262.0B |
| `newsfeed` | 18.1KB | 18.4KB | +365.0B |
| `observability` | 52.3KB | 52.7KB | +426.0B |
| `osquery` | 13.4KB | 13.6KB | +124.0B |
| `rollup` | 40.8KB | 33.8KB | -7.0KB |
| `screenshotMode` | 3.2KB | 3.5KB | +366.0B |
| `share` | 71.3KB | 71.8KB | +599.0B |
| `snapshotRestore` | 40.6KB | 40.9KB | +370.0B |
| `visTypeTagcloud` | 15.6KB | 14.4KB | -1.2KB |
| total |  |  | +10.5KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/012bb4490b5598790bc88fd112ab1c3d006a9a61) | [after](https://github.com/elastic/kibana/commit/38e0c5003901335ed24c459c0fea99e709a5e458) | diff |
| --- | --- | --- | --- |
| `core` | 2275 | 2276 | +1 |
| `data` | 3827 | 3831 | +4 |
| `esUiShared` | 88 | 90 | +2 |
| `expressions` | 1894 | 1896 | +2 |
| `fleet` | 1096 | 1101 | +5 |
| `indexPatternManagement` | 45 | 0 | -45 |
| `lens` | 172 | 174 | +2 |
| `management` | 38 | 39 | +1 |
| `observability` | 195 | 198 | +3 |
| `screenshotMode` | 21 | 22 | +1 |
| `share` | 67 | 85 | +18 |
| total |  |  | -6 |

#### async chunk count
| id | [before](https://github.com/elastic/kibana/commit/012bb4490b5598790bc88fd112ab1c3d006a9a61) | [after](https://github.com/elastic/kibana/commit/38e0c5003901335ed24c459c0fea99e709a5e458) | diff |
| --- | --- | --- | --- |
| `discover` | 7 | 8 | +1 |
| `lens` | 2 | 3 | +1 |
| `observability` | 5 | 7 | +2 |
| `osquery` | 3 | 4 | +1 |
| `visTypeTagcloud` | 1 | 2 | +1 |
| total |  |  | +6 |

#### References to deprecated APIs
| id | [before](https://github.com/elastic/kibana/commit/012bb4490b5598790bc88fd112ab1c3d006a9a61) | [after](https://github.com/elastic/kibana/commit/38e0c5003901335ed24c459c0fea99e709a5e458) | diff |
| --- | --- | --- | --- |
| `indexPatternManagement` | 68 | 0 | -68 |
| `observability` | 16 | 12 | -4 |
| `stackAlerts` | 101 | 104 | +3 |
| total |  |  | -69 |


</details>

### History
* :green_heart: [Build #130774](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130774/) succeeded f1a1a3e42a7bf52cf25018420a49a4fcf5934df6
* :green_heart: [Build #130470](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130470/) succeeded a25084fb95c1f49e00f69fc2e55be2631269ca85
* :broken_heart: [Build #130421](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130421/) failed bfe2dd3659a5ac39e7be55778235611374823fda

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130421/","number":"130421","commit":"bfe2dd3659a5ac39e7be55778235611374823fda"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130470/","number":"130470","commit":"a25084fb95c1f49e00f69fc2e55be2631269ca85"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/130774/","number":"130774","commit":"f1a1a3e42a7bf52cf25018420a49a4fcf5934df6"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/131537/","number":"131537","commit":"38e0c5003901335ed24c459c0fea99e709a5e458"}]}
PIPELINE-->
